tests/unit/amo/components/TestCollectionAddAddon.js

```diff
@@ -160,16 +180,73 @@ describe(__filename, () => {
     expect(root.find(Notice)).toHaveLength(1);
     expect(root.find(Notice).children()).toHaveText('Added to collection');
 
-    expect(root.instance().props.uiState.addonWasAdded).toEqual(true);
+    expect(root.instance().props.uiState.completedAddonAction).toEqual(
+      addonAddedAction,
+    );
+
+    // Trigger the setTimeout behavior.
+    clock.tick(MESSAGE_RESET_TIME);
+
+    applyUIStateChanges({ root, store });
+    expect(root.instance().props.uiState.completedAddonAction).toEqual(null);
+    expect(root.find(Notice)).toHaveLength(0);
+  });
+
+  it('displays a notification for 5 seconds after an add-on has been removed', () => {
+    const { store } = dispatchSignedInUser({
+      username: signedInUsername,
+    });
+    const root = render({ setTimeout: window.setTimeout, store });
+
+    expect(root.find(Notice)).toHaveLength(0);
+
+    _addonRemovedFromCollection({ root, store });
+
+    expect(root.find(Notice)).toHaveLength(1);
+    expect(root.find(Notice).children()).toHaveText('Removed from collection');
+
+    expect(root.instance().props.uiState.completedAddonAction).toEqual(
+      addonRemovedAction,
+    );
 
     // Trigger the setTimeout behavior.
     clock.tick(MESSAGE_RESET_TIME);
 
     applyUIStateChanges({ root, store });
-    expect(root.instance().props.uiState.addonWasAdded).toEqual(false);
+    expect(root.instance().props.uiState.completedAddonAction).toEqual(null);
     expect(root.find(Notice)).toHaveLength(0);
   });
 
+  it('clears the errorHandler when an add-on is added', () => {
+    const errorHandler = createStubErrorHandler(
+      new Error('Unexpected API error'),
+    );
+    errorHandler.clear = sinon.spy();
```

Comment:
It would be better to use the sinon feature for this so that we'll see an error in case `clear()` gets renamed in a refactor.

```js
const clearStub = sinon.stub(errorHandler, 'clear');
...
sinon.assert.called(clearStub);
```